support-models/src/main/scala/com/gu/support/workers/states/FailureHandlerState.scala

```diff
@@ -13,8 +13,8 @@ case class FailureHandlerState(
   giftRecipient: Option[GiftRecipient],
   redemptionData: Option[RedemptionData],
   product: ProductType,
-  paymentFields: Option[PaymentFields], //Will be present if CreatePaymentMethod failed
-  paymentMethod: Option[PaymentMethod], //Will be present if anything after CreatePaymentMethod failed
+  paymentFields: PaymentDetails[PaymentFields], //Will be present if CreatePaymentMethod failed
+  paymentMethod: PaymentDetails[PaymentMethod], //Will be present if anything after CreatePaymentMethod failed
```

Comment:
I thihnk these are technically `Option[PaymentDetails[PaymentFields]]` and `Option[PaymentDetails[PaymentMethod]]` as the two optionalities mean different things

Reply:
Yes! Well spotted.